Source/WebSocket_Windows.cpp

```diff
@@ -11,121 +11,98 @@ namespace UrlLib
 {
     using namespace winrt;
 
-    class WebSocket::WSImpl : public WSImplBase
+    class WebSocket::Impl : public ImplBase
     {
     public:
+        Impl(std::string url, std::function<void()> onOpen, std::function<void()> onClose, std::function<void(std::string)> onMessage, std::function<void()> onError)
+        : ImplBase(url, onOpen, onClose, onMessage, onError)
+        {
+        }
         
         void Send(std::string message)
         {
-            if (m_readyState == ReadyState::Closed ||
-                m_readyState == ReadyState::Closing ||
-                m_readyState == ReadyState::Connecting)
-            {
-                error_callback();
-                throw std::runtime_error{"WebSocket is not Open"};
-            }
-
             try
             {
                 Windows::Storage::Streams::DataWriter dataWriter{m_webSocket.OutputStream()};
                 dataWriter.WriteString(winrt::to_hstring(message));
 
                 arcana::create_task<std::exception_ptr>(dataWriter.StoreAsync())
-                    .then(arcana::inline_scheduler, arcana::cancellation::none(), [this, dataWriter](int)
+                    .then(arcana::inline_scheduler, m_cancellationSource, [this, dataWriter](int)
                     {
                         dataWriter.DetachStream();
+                    })
+                    .then(arcana::inline_scheduler, m_cancellationSource, [this](const arcana::expected<void, std::exception_ptr>& result)
```

Comment:
This is will execute on the wrong thread. We need to get back on to the JsRuntime thread to throw an error. Use JsRuntimeScheduler to do this.